lib/dispatcher/dispatcher.js

```diff
@@ -14,6 +17,28 @@ class Dispatcher extends EventEmitter {
   destroy () {
     throw new Error('not implemented')
   }
+
+  compose (...args) {
+    // So we handle [interceptor1, interceptor2] or interceptor1, interceptor2, ...
+    const interceptors = Array.isArray(args[0]) ? args[0] : args
+    let dispatcher = this
+    for (const interceptor of interceptors) {
+      if (interceptor == null) {
+        continue
+      }
+
+      if (typeof interceptor !== 'function') {
+        throw new Error('invalid interceptor')
+      }
+
+      dispatcher = interceptor(dispatcher) ?? dispatcher
```

Comment:
Can you clarify why you are doinng the `??` here? What use case are you trying to support?

As an added benefit, we could support a "shortcut" if the function returns a function `dispatch`, and we automatically add the `close` and `destroy`.

Reply:
Hmm, good point; I believe I brought similar point on the previous PR, overall by doing this we allow implementers to branch the dispatcher chain, which can cause undesired side-effects. Better be safer than sorry and if the compose callback returns either `null` or `undefined` we just ignore it (add a composed validation that checks for a `dispatcher-like` object).

Following the `dispatcher-like`, I'd say that if we introduce this pattern would be better to have some limitations to avoid misunderstanding and constant issue raising.
Just validating the `dispatch` is there, along with `close` and `destroy` should be good to go, but I wouldn't attache automatically `close` and `destroy` as those potentially can cause side-effects users are not aware of.

The last being said, wdyt about validating the `dispatcher-like` shape and documenting it?

Reply:
Agreed